### PR TITLE
WD-11665 - refactor: remove nested components in CharmActionsPanel

### PR DESCRIPTION
**Pull request by vladimir-cucu**
## Done

- Removed nested components in `CharmActionsPanel`.

## Details

- https://warthogs.atlassian.net/browse/WD-11665

### CODE REVIEW

File: src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    const cancelButton = screen.getByRole("button", {
+      name: Label.CANCEL_BUTTON,
+    });
+    expect(cancelButton).toBeInTheDocument();
+    expect(
+      screen.getByRole("button", { name: Label.CONFIRM_BUTTON }),
+    ).toBeInTheDocument();
+    expect(
+      screen.getByText("APPLICATION COUNT (UNIT COUNT)"),
+    ).toBeInTheDocument();
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx:130]: I'm not sure what we gain by testing this. What fail condition are you trying to catch?

File: src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    expect(
+      screen.getByRole("dialog", { name: "Run stdout?" }),
+    ).toBeInTheDocument();
+    const cancelButton = screen.getByRole("button", {
+      name: Label.CANCEL_BUTTON,
+    });
+    expect(cancelButton).toBeInTheDocument();
+    expect(
+      screen.getByRole("button", { name: Label.CONFIRM_BUTTON }),
+    ).toBeInTheDocument();
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx:127]: Probably don't gain much by testing that this button exists.

File: src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    await userEvent.click(
+      screen.getByRole("button", { name: Label.CONFIRM_BUTTON }),
+    );
+    expect(mockSetConfirmType).toHaveBeenCalledWith(null);
+    const call = executeActionOnUnitsSpy.mock.calls[0];
+    expect(call[1]).toBe("add-disk");
+    expect(call[0]).toEqual(["ceph-0", "ceph-1"]);
+    expect(call[2]).toEqual({});
+    expect(await screen.findByText(Label.ACTION_ERROR)).toBeInTheDocument();
+    expect(mockOnRemovePanelQueryParams).toHaveBeenCalledOnce();
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx:199]: This test should just test that part that we're concerned with (the other lines are captured by the success case above).
```suggestion
    expect(await screen.findByText(Label.ACTION_ERROR)).toBeInTheDocument();
```

File: src/panels/ActionsPanel/CharmActionsPanel/CharmActionsPanel.tsx [hunk 4]
```diff
@@ -210,7 +122,18 @@ export default function CharmActionsPanel({
...
         ))}
-        {generateConfirmationModal()}
+        <ConfirmationDialog
+          confirmType={confirmType}
+          selectedAction={selectedAction}
+          selectedApplications={selectedApplications}
+          setConfirmType={setConfirmType}
+          selectedActionOptionValue={
+            // {} is a placeholder value, as ConfirmationDialog will not render
+            // if selectedAction is undefined.
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/CharmActionsPanel.tsx:132]: I think if we're needing to do this to get around typing issues, then it's probably a sign that we need to move the `selectedAction` check into this component (`{selectedAction ? <ConfirmationDialog ... /> : null}`).

File: src/panels/ActionsPanel/CharmActionsPanel/index.ts [hunk 5]
```diff
@@ -1,2 +1,2 @@
 export { default } from "./CharmActionsPanel";
-export { Label as CharmActionsPanelLabel } from "./types";
+export { Label as CharmActionsPanelLabel } from "./ConfirmationDialog/types";
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/index.ts:2]: This export should be in `ConfirmationDialog/index.ts` as `ConfirmationDialogLabel` and this component should export the label defined in `./types`.

File: src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  setConfirmType,
+  selectedActionOptionValue,
+  onRemovePanelQueryParams,
+}: Props): JSX.Element | null => {
+  const { Portal } = usePortal();
+  const { userName, modelName } = useParams();
+  const sendAnalytics = useAnalytics();
+  const executeActionOnUnits = useExecuteActionOnUnits(userName, modelName);
+
+  if (selectedAction && confirmType === ConfirmType.SUBMIT) {
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.tsx:83]: `selectedAction` is always defined now.

File: src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        selectedAction=""
+        selectedApplications={[]}
+        setConfirmType={vi.fn()}
+        selectedActionOptionValue={{}}
+        onRemovePanelQueryParams={vi.fn()}
+      />,
+    );
+    expect(container.children.length).toBe(1);
+    expect(container.firstChild).toBeEmptyDOMElement();
+  });
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx:82]: This test is no longer required as `selectedAction` is always defined.

**Review by huwshimi — APPROVED** [review]
Just two tiny changes.

File: src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+      }),
+    });
+  });
+
+  afterEach(() => {
+    vi.resetModules();
+    vi.restoreAllMocks();
+  });
+
+  it("should return null if there is no selected action", () => {
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.test.tsx:67]:
```suggestion
  it("should return null if there the confirm type is not 'submit'", () => {
```

File: src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+  selectedActionOptionValue,
+  onRemovePanelQueryParams,
+}: Props): JSX.Element | null => {
+  const { Portal } = usePortal();
+  const { userName, modelName } = useParams();
+  const sendAnalytics = useAnalytics();
+  const executeActionOnUnits = useExecuteActionOnUnits(userName, modelName);
+
+  if (confirmType === ConfirmType.SUBMIT) {
+    console.log("here");
```
Comment (huwshimi) [inline src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.tsx:84]:
```suggestion
```

**Review by huwshimi — APPROVED** [review]

File: src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.tsx (hunk 9 above)
Comment (vladimir-cucu) [reply src/panels/ActionsPanel/CharmActionsPanel/ConfirmationDialog/ConfirmationDialog.tsx:84]: Ooops :) Thanks for the review!

**Pull request merged**